lambdas/sftp-letters/src/app/poll.ts

```
@@ -0,0 +1,115 @@
+  ) {}
+
+  private async copyFolder(
+    sftpClient: SftpClient,
+    copyPath: string,
+    pastePath: string
+  ) {
+    this.logger.info({
+      description: 'Copying folder',
+      copyPath,
+      pastePath,
+    });
+    const isDir = (await sftpClient.exists(copyPath)) === 'd';
+
+    if (!isDir) {
+      this.logger.info(`Path '${copyPath}' does not exist`);
+      return;
+    }
+
+    const sftpFiles = await sftpClient.list(copyPath);
+
+    for (const sftpFile of sftpFiles) {
+      if (sftpFile.type === 'd') {
+        await this.copyFolder(
+          sftpClient,
+          `${copyPath}/${sftpFile.name}`,
+          `${pastePath}/${sftpFile.name}`
+        );
+      }
+
+      if (sftpFile.type === '-') {
+        await this.copyFile(
+          sftpClient,
+          `${copyPath}/${sftpFile.name}`,
+          `${pastePath}/${sftpFile.name}`
+        );
+      }
+    }
+  }
+
+  private async validateFile(buffer: Buffer) {
+    return (
+      buffer.byteLength > 0 &&
+      buffer.byteLength < MAX_SIZE_BYTES &&
+      this.isBufferPdf(buffer)
+    );
+  }
+
+  isBufferPdf(buffer: Buffer): boolean {
+    const pdfBufferStart = [0x25, 0x50, 0x44, 0x46]; // '%PDF'
+    return pdfBufferStart.every((byte, index) => buffer[index] === byte);
+  }
+
+  private async copyFile(
+    sftpClient: SftpClient,
+    copyPath: string,
+    pastePath: string
+  ) {
+    this.logger.info({ description: 'Copying file', copyPath, pastePath });
+    try {
+      const data = (await sftpClient.get(copyPath)) as Buffer;
+
+      const fileValidation = await this.validateFile(data);
+
+      if (!fileValidation) {
+        this.logger.error('PDF file failed validation', {
+          copyPath,
+          pastePath,
+        });
+        return;
+      }
+
+      await this.s3Repository.putRawData(data, pastePath);
+
+      await sftpClient.delete(copyPath);
+    } catch (error) {
+      this.logger.error('Failed to copy file', { copyPath, pastePath, error });
```

The error will be lost here. I would do `this.logger..child({description:'...',copyPath, pastePath})error(error)`